pyro/distributions/von_mises.py

```diff
@@ -52,25 +64,54 @@ class VonMises(TorchDistribution):
     See :class:`~pyro.distributions.VonMises3D` for a 3D cartesian coordinate
     cousin of this distribution.
 
-    Currently only :meth:`log_prob` is implemented.
-
     :param torch.Tensor loc: an angle in radians.
     :param torch.Tensor concentration: concentration parameter
     """
     arg_constraints = {'loc': constraints.real, 'concentration': constraints.positive}
     support = constraints.real
+    has_rsample = True
 
     def __init__(self, loc, concentration, validate_args=None):
         self.loc, self.concentration = broadcast_all(loc, concentration)
         batch_shape = self.loc.shape
         event_shape = torch.Size()
+
+        # Moments
+        self._variance = 1 - (_log_modified_bessel_fn(self.concentration, order=1) -
+                              _log_modified_bessel_fn(self.concentration, order=0)).exp()
+
+        # Parameters for sampling
+        tau = 1 + (1 + 4 * self.concentration ** 2).sqrt()
+        rho = (tau - (2 * tau).sqrt()) / (2 * self.concentration)
+        self._proposal_r = (1 + rho ** 2) / (2 * rho)
+
         super(VonMises, self).__init__(batch_shape, event_shape, validate_args)
 
     def log_prob(self, value):
         log_prob = self.concentration * torch.cos(value - self.loc)
-        log_prob = log_prob - math.log(2 * math.pi) - _log_modified_bessel_fn_0(self.concentration)
+        log_prob = log_prob - math.log(2 * math.pi) - _log_modified_bessel_fn(self.concentration, order=0)
         return log_prob
 
+    def rsample(self, sample_shape=torch.Size()):
```

Comment:
I believe rejection samplers do not admit reprametrization, so you should instead
```py
has_rsample = False

@torch.no_grad()
def sample(self, sample_shape=torch.Size()):
    ...
```
We can try to wrap this in a `Rejector` in a follow-up PR if you're interested.